src/base/bittorrent/session.cpp

```diff
@@ -2368,6 +2368,36 @@ void Session::setIPFilterFile(QString path)
     }
 }
 
+void Session::setBannedIPs(const QStringList &newList)
+{
+    if (newList == m_bannedIPs)
+        return; // do nothing
+    // malformed IP strings will be silently filtered out
+    QStringList filteredList;
+    foreach (const QString &ip, newList) {
+        boost::system::error_code ec;
+        libt::address::from_string(ip.toLatin1().constData(), ec); // only error code needed
+        if (!ec) {
+            filteredList << ip;
+        }
+        else {
+            Logger::instance()->addMessage(
+                        tr("%1' is not a valid IP address and was rejected while applying the list of banned addresses.")
+                        .arg(ip)
+                        , Log::INFO);
+        }
+    }
+    // store to session settings
+    m_bannedIPs = filteredList;
+    m_IPFilteringChanged = true;
+    configureDeferred();
```

Comment:
But my proposal doesn't account for one corner-case bug:
1. User has manually banned an IP that is also included in the IP filter file.
2. Then he deletes it from the manually banned IPs list.

In my proposal step 2 would result in allowing that IP even though the ip_filter includes it(until the user restarts the application).
So would it be wise, to reparse the ip_filter file, add the bannedIPs and set that whole list in libtorrent, discarding the old one?